stm32 adc define the internal voltage reference

FRASTM opened the pull request:
This change is to define the ref_internal as a pvalue of the adc_driver_api
This Analog reference voltage for the ADC is given by the board AVDD to the VREF+ input of the MCU
It is usually connected to the VDD supply line.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
drivers: adc: stm32 adc API has a valid internal reference

extend the stm32 adc api to set the ref_internal of 3.3V

Signed-off-by: Francois Ramu <francois.ramu@st.com>

erwango commented on drivers/adc/adc_stm32.c:
```diff
@@ -811,6 +812,11 @@ static void adc_stm32_setup_channels(const struct device *dev, uint8_t channel_i
 	}
+
+	if (config->has_vbat_channel &&
+		__LL_ADC_CHANNEL_TO_DECIMAL_NB(ADC_CHANNEL_VBAT) == channel_id) {
```
To respect API (cf similar change done in https://github.com/zephyrproject-rtos/zephyr/pull/43857)
```suggestion
		__LL_ADC_CHANNEL_TO_DECIMAL_NB(LL_ADC_CHANNEL_VBAT) == channel_id) {
```

erwango commented on drivers/adc/adc_stm32.c:
```diff
@@ -811,6 +812,11 @@ static void adc_stm32_setup_channels(const struct device *dev, uint8_t channel_i
 	}
+
+	if (config->has_vbat_channel &&
+		__LL_ADC_CHANNEL_TO_DECIMAL_NB(ADC_CHANNEL_VBAT) == channel_id) {
```
U5 also has LL_ADC_CHANNEL_VBAT and LL_ADC_CHANNEL_VBAT_ADC4 which looks like a different channel.

FRASTM replied:
removing this part from that PR

erwango reviewed (changes requested):
Miss dts & bindings updates

FRASTM commented:
change this PR to only have the **ref_internal** setting

FRASTM committed:
dts: binding: stm32adc set the VREf as a property

Add the Vref reference voltage in the DTS, so that the adc driver
can retrieve the value for conversion.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

erwango commented on dts/bindings/adc/st,stm32-adc.yaml:
```diff
@@ -27,6 +27,11 @@ properties:
+    vref_mv:
+      type: int
+      default: 3300
+      description: Indicates the reference voltage of the ADC in mV.
```
Required:false

FRASTM replied:
adding
`       required: false`

ycsin commented:
@FRASTM @erwango is this something that should be backported, if it fixes anything?

erwango commented:
@ycsin Please don't add backport label on all PRs. As already explained, only bugfixes should be backported (and not all of them). This is not the case here. Please see discussions
https://github.com/zephyrproject-rtos/zephyr/discussions/43561 and  https://github.com/zephyrproject-rtos/zephyr/issues/43335

ycsin commented:
@erwango I actually think that this PR would fit into "Usability Bug Fixes and improvements", since it will make the [adc sample](https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/drivers/adc/src/main.c#L94) work.

erwango commented:
> @erwango I actually think that this PR would fit into "Usability Bug Fixes and improvements", since it will make the [adc sample](https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/drivers/adc/src/main.c#L94) work

So indeed, "Usability Bug Fixes and improvements" is the part which is being discussed and that some would like to prevent being merged. We should focus on keeping the branch stable as it should be maintained for next 2 years. So we should minimize the changes and put prio on actual fw **bugs**.
